resources/js/app.js

```
@@ -391,6 +391,38 @@ $(".hr_round_guide").on("click", ".edit-guide", function () {
 	container.find(".btn-guide, .guide-container").toggleClass("d-none");
 });
 
+$(document).ready(function () {
+	$("#addNewSegmentForm").submit(function (e) {
+		e.preventDefault();
+		let form = $("#addNewSegmentForm");
+		$("#createNewSegment").on("hidden.bs.modal", function () {
+			$(this).find("form").trigger("reset");
+		});
+	
+		$.ajax({
+			type: form.attr("method"),
+			url: form.attr("action"),
+			data: form.serialize(),
+			success:function (response) {
+				$("#createNewSegment").modal("hide");
+				$("#createNewSegment").on("hidden.bs.modal", function (e)
+				 {
+					$("#segmentsuccess").toggleClass("d-none");
+					$("#segmentsuccess").fadeToggle(6000);
+				});
+			},
+			error: function(response){
+				if(response.responseJSON.errors.name) {
+					$("#segmentNameError").removeClass("d-none");
+				} 
+				if(response.responseJSON.errors.rounds) {
+					$("#roundNameError").removeClass("d-none");
+				}
```

Instead of showing and hiding the error div you should set the text of the div based on the errors message that you get from `response.responseJSON.errors.`

This way we can show different messages in future when we add more fields, without updating the logic. 

**Action items**
Just create an empty div and show the errors messages received in the error in that div using jS